src/ssl.c

```diff
@@ -12946,6 +12947,26 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
         ctx->cm               = str->cm;
         ctx->x509_store.cm    = str->cm;
 
+        /* wolfSSL_CTX_set_cert_store() (this function) associates str with the
+         * wolfSSL_CTX. It is clear that this is a TLS use case which means we
+         * should move all the certs, if any, into the CertMgr and set
+         * str->certs to NULL as that will allow the certs to be properly
+         * processed. */
+        if (str->certs != NULL) {
+            while (wolfSSL_sk_X509_num(str->certs) > 0) {
+                int ret;
+                x = wolfSSL_sk_X509_pop(str->certs);
+                ret = X509StoreAddCa(str, x, WOLFSSL_USER_CA);
+                if (ret != WOLFSSL_SUCCESS) {
+                    WOLFSSL_MSG("Error adding CA certificate to store");
+                    wolfSSL_X509_free(x); /* Free the certificate to avoid memory leaks */
+                    break; /* Exit the loop on failure */
+                }
+            }
+            wolfSSL_sk_X509_pop_free(str->certs, NULL);
```

Comment:
Why does the `certs` stack need free'd and emptied here? Is it being converted over to a WOLFSSL_CTX cert store, and that future calls to wolfSSL_X509_STORE_add_cert would then load `certs` as a CA?

Reply:
Also it would be worth checking if this behavior conforms to the expected OpenSSL compat behavior. Attempting untrusted vs trusted self signed certificate loads after the store is set to the WOLFSSL_CTX struct.